yeat/config/config.py

```diff
@@ -48,23 +48,26 @@ def check_valid_keys(self, observed_keys, expected_keys):
             message = f"Found unsupported configuration key(s) '{key_str}'"
             raise AssemblyConfigError(message)
 
-    def create_sample_and_assembly_objects(self):
+    def create_sample_objects(self):
+        samples = {}
         for label, sample in self.config["samples"].items():
-            self.samples[label] = Sample(label, sample)
+            samples[label] = Sample(label, sample)
+        return samples
+
+    def create_assembly_objects(self):
+        assemblies = {}
         for label, assembly in self.config["assemblies"].items():
-            self.assemblies[label] = Assembly(label, assembly, self.threads)
+            assembly["samples"] = self.get_sample_objects(assembly["samples"])
+            assemblies[label] = Assembly(label, assembly, self.threads, self.bandage)
```

Comment:
This here is new. To make the final file calculation possible in the Assembly class, we need to pass in their sample objects containing specific metadata such as read type and its quality.

Note: readtype is needed for the snakemake output directory structure. (see diagram in the PR's description.)